src/utils/beacon/decodePayload.ts

```diff
@@ -0,0 +1,52 @@
+import { hex2buf } from "@taquito/utils";
+
+const PAYLOAD_PADDING = "0501";
+
+/**
+ * Decodes a sign request payload string.
+ *
+ * @param payload - The payload to decode.
+ * @returns The decoded string, or the original payload if decoding fails.
+ * @see {@link https://taquito.io/docs/signing/} for more info.
+ */
+export const decodePayload = (payload: string): string => {
+  let index = 0;
+
+  if (payload.startsWith(PAYLOAD_PADDING)) {
+    index = 4;
+
+    while (index < payload.length && payload[index] === "0") {
+      index += 1;
+    }
+
+    let rawBytesLength = "";
+    let bytesLength = 0;
+    const paddingLength = index;
+
+    while (index < payload.length && payload[index] !== "0") {
+      rawBytesLength += payload[index];
```

Comment:
`rawBytesLength.length` seems a bit confusing to me

Please consider renaming vars to maybe something like

`rawBytesLength` => `payloadByteLengthRaw`
`bytesLength` => `payloadByteLength`